beacon-chain/sync/backfill/worker.go

```diff
@@ -56,19 +72,55 @@ func (w *p2pWorker) handle(ctx context.Context, b batch) batch {
 	for i := range vb {
 		bdl += vb[i].SizeSSZ()
 	}
-	backfillBatchApproximateBytes.Add(float64(bdl))
-	log.WithField("dlbytes", bdl).Debug("backfill batch bytes downloaded")
+	backfillBlocksApproximateBytes.Add(float64(bdl))
+	log.WithFields(b.logFields()).WithField("dlbytes", bdl).Debug("backfill batch block bytes downloaded")
 	b.results = vb
+	b.bs, err = newBlobSync(cs, blobRetentionStart, b.results, w.nbv, w.bfs)
+	if err != nil {
+		return b.withRetryableError(err)
+	}
+	if b.blobsNeeded() > 0 {
+		return b.withState(batchBlobSync)
+	}
+	return b.withState(batchImportable)
+}
+
+func (w *p2pWorker) handleBlobs(ctx context.Context, b batch) batch {
+	b.blobPid = b.busy
+	start := time.Now()
+	// we don't need to use the response for anything other than metrics, because blobResponseValidation
+	// adds each of them to a batch AvailabilityStore once it is checked.
+	blobs, err := sync.SendBlobsByRangeRequest(ctx, w.c, w.p2p, b.blobPid, w.cm, b.blobRequest(), b.blobResponseValidator(), blobValidationMetrics)
+	if err != nil {
+		b.bs = nil
+		return b.withRetryableError(err)
+	}
+	dlt := time.Now()
+	backfillBatchTimeDownloadingBlobs.Observe(float64(dlt.Sub(start).Milliseconds()))
+	if len(blobs) > 0 {
+		// All blobs are the same size, so we can compute 1 and use it for all in the batch.
+		sz := blobs[0].SizeSSZ() * len(blobs)
+		backfillBlobsApproximateBytes.Add(float64(sz))
+		log.WithFields(b.logFields()).WithField("dlbytes", sz).Debug("backfill batch blob bytes downloaded")
+	}
+	if b.blobsNeeded() > 0 {
```

Comment:
not possible the way it's currently written but good idea to check it for future proofing.

Reply:
The only way to get into this func is if state is `batchBlobSync`, and I've changed the code so the only way to get into that state and set the value for `batchh.results` is to also set the `batch.bs` via `withResults`. I think that makes calling a func that assumes batch.bs is not nil fairly safe. This can be improved with some more refactoring after I get the clock initialization overhaul done.